### PR TITLE
COMMERCE-1639 refactoring for managing product options

### PR DESCRIPTION
**Pull request by luca-pellizzon**

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test
Forgot to update private branch :'(

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test
never completed yesterday I guess

**Comment by ethanbustad** [comment]
@luca-pellizzon looks like most of the `-web` modules need a project dependency for `commerce-product-api` because of the baseline change.

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
Hey @alecsloan , since @gcmznt and I did lots of fixes to
commerce-theme-minium-impl\src\main\resources\META-INF\resources\render\view.jsp

Can you check that your changes introduced with "COMMERCE-1731 Fix the price(s) rendered when there's CPOptions" still work after my rebase?

Thanks

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by alecsloan** [comment]
Thanks @luca-pellizzon  I'll check it out

**Comment by alecsloan** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Review by ethanbustad — COMMENTED** [review]
@luca-pellizzon this mostly looks great, and I think it'll help solve an issue I've been working on with 7.2 compatibility. Just one thing to bring up: for the new `renderOptions` and `renderPublicStoreOptions` methods, they look like _almost_ exact copies of the methods that were already there. What do you think of having the old methods call the new ones, with default values passed in?

File: commerce-theme-minium/commerce-theme-minium-impl/src/main/resources/META-INF/resources/render/view.jsp [hunk 1]
```diff
@@ -20,14 +20,24 @@
...
 
 long cpDefinitionId = cpCatalogEntry.getCPDefinitionId();
 
-String addToCartId = PortalUtil.generateRandomKey(request, "add-to-cart");
+String addToCartId = PortalUtil.generateRandomKey(request, "addToCart");
+String ddmFormContainerId = PortalUtil.generateRandomKey(request, "formContainer");
 String galleryId = PortalUtil.generateRandomKey(request, "gallery");
 %>
 
+<script>
```
Comment (marco-leo) [inline commerce-theme-minium/commerce-theme-minium-impl/src/main/resources/META-INF/resources/render/view.jsp:33]: @ethanbustad I don't like this

Comment (ethanbustad) [reply commerce-theme-minium/commerce-theme-minium-impl/src/main/resources/META-INF/resources/render/view.jsp:33]: Don't like the `<script>` tag here? Or defining global variables at all?

**Comment by luca-pellizzon** [comment]
@ethanbustad Updated CPInstanceHelperImpl so that the old method call the new one.
Didn't do the same regarding CPContentHelperImpl because in order to retrieve the "default value" we need to do pretty much the same operations as before, so not simplifying anything in this case

**Comment by ethanbustad** [comment]
ci:test
Tests may pass now that https://github.com/liferay/com-liferay-commerce/pull/1436 is merged in.

**Comment by luca-pellizzon** [comment]
ci:test

**Review by ethanbustad — APPROVED** [review]

File: commerce-theme-minium/commerce-theme-minium-impl/src/main/resources/META-INF/resources/render/view.jsp (hunk 1 above)
Comment (marco-leo) [reply commerce-theme-minium/commerce-theme-minium-impl/src/main/resources/META-INF/resources/render/view.jsp:33]: defining global variables

Comment (gcmznt) [reply commerce-theme-minium/commerce-theme-minium-impl/src/main/resources/META-INF/resources/render/view.jsp:33]: This global variables are a fix for the navigation. they are used in the `aui:script` at line 99.
The `script` tag is needed for timing reasons, we need these variables not cached and executed immediately.

So if we want to remove global variables we need to find other ways to solve this problem

**Comment by gcmznt** [comment]
Any news on this PR, I need this to update the gallery component @marco-leo

**Comment by ethanbustad** [comment]
@gcmznt this would at least need to be rebased (@luca-pellizzon). But other than that I think Marco's ignoring it because he doesn't like it :)

**Comment by luca-pellizzon** [comment]
@gcmznt @ethanbustad talked with Marco, and the problem is that we need to test it using the simple renderer. Probably as it is right now it only works with minium

**Comment by gcmznt** [comment]
I am working on the product detail page, and for me it can be useful to use this refactored javascript. Do you think I can just move the JavaScript code in my branch?

**Comment by luca-pellizzon** [comment]
@gcmznt  You can try to start directly from this PR, as you can see all the conflicts are just build.gradle so you can manage it quite easily.
BTW I'm rebasing it, so give me some minutes and it will be ready to use 😃

**Comment by marco-leo** [comment]
closing as was solver refactoring the options

**Pull request closed without merging**